### ISSUE TITLE
Make fitscheck check compliance by default

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#6548]
The ``fitscheck`` command only checks checksums by default, but has a ``--compliance`` flag to check for compliance with the FITS standard. As a user I had assumed for several years that ``fitscheck`` defaulted to doing compliance checking too. I think we should actually just turn on compliance checking by default and add a flag to disable it, since I think other users will likely be expecting the same.

**Comment by saimn** [comment]
I agree: the fact that this script is meant to verify (and fix) checksums but can also optionally verify headers is confusing (going to a specific thing to a more general one). I would make more sense to me to have have a script that verifies headers, and checksum if there are present, and optionally can fix the headers and checksum.
